design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+the resource at a provider. Crossplane takes ownership of the resource, starts acting as the source of truth, and
+ensures configuration matches the desired state.
+
+Sometimes, we want to "observe" an existing resource without taking ownership of it. This could be useful in several
+scenarios, which could be grouped as follows:
+
+- Referencing existing resources without managing them
+  - In your composition, you want to reference network resources like VPC and subnets that are managed by another tool
+  or team.
+- Fetching data from existing resources
+  - You need information about an existing VPC, such as its CIDR range and the subnets it contains.
+- Gradual migration of existing/legacy infrastructure to Crossplane
+  - You have existing infrastructures managed by Terraform, and you want to migrate them gradually to Crossplane.
+  - You have a legacy infrastructure that you want to migrate to Crossplane, but you want to test the managed resources
+  before taking ownership of the underlying resources.
+  - For an existing resource, you don’t want to give some initial configuration that might override the actual
+  configuration. You want to late-initialize all fields, including the ones that would be required otherwise.
+- Only observing some fields after the initial creation
+  - You want to create an EKS Node Group with a scaling configuration where you configured an initial desired size.
+  After the creation, you want to ignore changes in the size which is now being controlled by the cluster autoscaler.
+
+Currently, Crossplane does not have a built-in way of observing resources without taking ownership of them. There are
+two workarounds used by the community as an interim solution for this gap:
+
+- Using the provider-terraform to observe resources with the help of Terraform data sources.
+- Wrapping resources with provider-kubernetes to observe resources managed by Crossplane but as a shared object between
+multiple Compositions.
+
+In this document, we aim to introduce a solution to observe resources with Crossplane without taking ownership of them.
+This would allow users to integrate existing cloud resources with the Crossplane ecosystem without giving full ownership.
+
+## Goals
+
+- Introduce a way to observe existing resources without taking ownership of them in Crossplane.
+- Enable seamless integration of existing cloud resources with the Crossplane ecosystem.
+- Allow users to fetch data from existing resources without giving full ownership.
+
+### Non-goals
+
+- Partially managing a resource by observing a subset of fields.
+
+## Proposal
+
+To support observing resources without taking ownership, we will introduce a new spec named `managementPolicy` to the
+Managed Resources. We will also deprecate the existing `deletionPolicy` in favor of the new spec since they will be
+controlling the same behavior; that is, how should the changes on the CR affect the external cloud resource.
```

Comment:
I wouldn't feel comfortable with naming this as `managementPolicy` if we don't deprecate `deletionPolicy` since they are quite relevant regarding what they are doing, that is, controlling CRUD behavior on external resources.

If this would be the way we would like to go, then I would rather use a parameter like `observeOnly: true` in spec or may be an annotation, `crossplane.io/observe-only: true`. TBH, I am not as excited as current proposal with these options 🙂 

FWIW, I don't believe the logic for handling both would be too complex, and don't see a big problem with it staying around for a while.

Reply:
Just to get it persisted here. 
As discussed in the community meeting, we may choose a proactive path and leverage a conversion webhook to drop the deletion policy by converting it to the corresponding management policy instead of leaving it as deprecated. This would require an API Version bump on affected resources (i.e., all managed resources) which we need to plan carefully, and we may consider dropping a couple of other deprecated fields altogether.